### PR TITLE
Bug 1127207 - [Text Selection] migrate touchcarettap handler to selectionstatechange

### PR DESCRIPTION
**Pull request by cctuan**
https://bugzilla.mozilla.org/show_bug.cgi?id=1127207

### CODE REVIEW

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=ac8fa67c07df)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8743aa8b087b)

File: apps/system/js/text_selection_dialog.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* global layoutManager, SettingsListener */
 'use strict';
 (function(exports) {
-  var DEBUG = false;
+  var DEBUG = true;
```
Comment (cctuan) [inline apps/system/js/text_selection_dialog.js]: set to false

**Comment by cctuan** [comment]
some typo also need to fixed.ex: buttom -> bottom

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=1af10077b46f)

File: apps/system/test/marionette/lib/text_selection.js [hunk 2]
```diff
@@ -88,7 +88,7 @@ TextSelection.prototype = {
  
   longPressByPosition: function(element, x, y) {
     // Add moveByOffset to prevent contextmenu event be fired.
-    this.actions.tap(element, x, y).wait(2).press(element, x, y).
+    this.actions.tap(element, x, y).wait(4).press(element, x, y).
```
Comment (mephisto41) [inline apps/system/test/marionette/lib/text_selection.js]: Should't change this?

File: apps/system/test/marionette/text_selection_test.js [hunk 3]
```diff
@@ -241,7 +241,22 @@ marionette('Text selection >', function() {
...
 
-      test('bug1119126 : Shortcut bubble should hide when system is resized',
+      // Enable the test til bug 1120750 is merged.
+      test('bug1120750 : Send out carets position for the short cut mode ',
+        function() {
+          fakeTextselectionApp.longPress('BugCenterInput');
+          var originalLocation = fakeTextselectionApp.textSelection.location;
+          fakeTextselectionApp.cut('BugCenterInput');
+
+          fakeTextselectionApp.longPress('BugBottomInput');
```
Comment (mephisto41) [inline apps/system/test/marionette/text_selection_test.js]: Use tap for appearing shortcut is enough. Use long press maybe wait too long time so the shortcut has been timeout.

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=960f1d98697d)

File: apps/system/js/text_selection_dialog.js [hunk 4]
```diff
@@ -121,7 +111,27 @@
...
+            evt.stopPropagation();
+
+            var detail = evt.detail.detail;
+            if (!detail) {
+              return;
+            }
+            this.debug('on receive selection change event');
+            this.debug(JSON.stringify(detail));
+
+            this._isSelectionVisible = detail.visible;
```
Comment (alivedise) [inline apps/system/js/text_selection_dialog.js]: Why do you need to specify here and L167 and L193?

File: apps/system/test/marionette/lib/faketextselectionapp.js [hunk 5]
```diff
@@ -30,7 +30,7 @@ FakeTextSelectionApp.Selector = Object.freeze({
   NonEditableNonSelectedDiv: '#noneditable-userselectnone',
   // bug.html
   BugCenterInput: '#bug-center-input',
-  BugButtomInput: '#bug-buttom-input',
+  BugButtomInput: '#bug-bottom-input',
```
Comment (alivedise) [inline apps/system/test/marionette/lib/faketextselectionapp.js]: BugButtomInput

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=1a38ef468b99)

**Pull request merged**